### DISCUSSION TITLE
How do I get "work" tags?

### DISCUSSION THREAD

**Discussion by InvisibleFunction** [beetbox/beets#5547] [Q&A]
I use LMS which just added support for files tagged with work. I looked at my files and I don't seem to have that tag, though a number do have "mb_workid"/"MUSICBRAINZ_WORKID". How do I get things tagged with "work"?

```
me@compy:/Volumes/Music/Music$ beet info -l The\ Milk\ Carton\ Kids/Christmas\ In\ A\ Minor\ Key/01\ The\ First\ Noel.flac  | grep -i work
          mb_parentworkid: 78bb65b9-29a1-414e-a503-673894d636a3
                mb_workid: 631091d0-feb9-4edd-ac43-95ab24b3bc23
               parentwork: Meet Me in St. Louis
      parentwork_disambig: 1944 film
parentwork_workid_current: 631091d0-feb9-4edd-ac43-95ab24b3bc23
                     work: The First Noël
            work_disambig:
me@compy:/Volumes/Music/Music$ beet info The\ Milk\ Carton\ Kids/Christmas\ In\ A\ Minor\ Key/01\ The\ First\ Noel.flac  | grep -i work
           mb_workid: 631091d0-feb9-4edd-ac43-95ab24b3bc23
me@compy:/Volumes/Music/Music$ metaflac --list The\ Milk\ Carton\ Kids/Christmas\ In\ A\ Minor\ Key/01\ The\ First\ Noel.flac | grep -i work
    comment[58]: MUSICBRAINZ_WORKID=631091d0-feb9-4edd-ac43-95ab24b3bc23
```

It looks like the info is in the db, but is not written to the files?

EDIT: I think I need to extend https://github.com/beetbox/mediafile , is that right?

**Comment by antonmosich** [comment]
I'd also be interested in this. Furthermore it would be really nice to be able to write the `parentwork` information into the work tag. I patched mediafile locally and successfully wrote the work tag. If someone could provide some idea on how to implement this I could try to do it.

**Reply by InvisibleFunction** [reply]
It looks like someone was [looking into writing a plugin](https://forums.slimdevices.com/forum/user-forums/logitech-media-server/1737175-lyrion-music-server-9-0-0-classical-works-functionality-support/page6#post1740488). I'm not really clear on how things should work. It seems like for classical I'd want to see the `parentwork` tag, but for other genres I'd want to see the `work` tag? And I guess LMS should know how to do that? Haven't read through everything on the forum, but hopefully someone is working on making this easyish.

**Reply by mirthestam** [reply]
> It looks like someone was [looking into writing a plugin](https://forums.slimdevices.com/forum/user-forums/logitech-media-server/1737175-lyrion-music-server-9-0-0-classical-works-functionality-support/page6#post1740488). I'm not really clear on how things should work. It seems like for classical I'd want to see the `parentwork` tag, but for other genres I'd want to see the `work` tag? And I guess LMS should know how to do that? Haven't read through everything on the forum, but hopefully someone is working on making this easyish.

Indeed, that was my post. I am currently working on the plugin, and there are also a few bug fixes needed in Beets itself. I will first submit these through a pull request. The basic functionality appears to be solid, but the plugin heavily relies on MusicBrainz tags and is currently focused on Lyrion due to its complexity. Once I have something worthwhile to share, I will post a message on the forum.
